### PR TITLE
cleanup: naming and README

### PR DESCRIPTION
**Pull request by lamping7**
* update README hierarchy
* alphabetize and fix Makefile
* add version to the name of the base and ros image

- [x] TODO: Update docker hub on file name changes once reviewed and good to merge

EDIT: only a cleanup of px4-dev.

### CODE REVIEW

**Comment by lamping7** [comment]
Additional discussion points:
- Keep raspi? We have armhf.
- Why is everything named "-dev"? It doesn't really distinguish it from anything else.
- Keep ROS2 ardent (it's past EOL - Dec 2018)

File: docker/px4-dev/Dockerfile_ros-kinetic [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # PX4 ROS development environment
 #
 
-FROM px4io/px4-dev-base-xenial:pr-base-ubuntu16
+FROM px4io/px4-dev-base-xenial:latest
```
Comment (lamping7) [inline docker/px4-dev/Dockerfile_ros-kinetic]: This will still need a proper tag eventually

**Comment by TSC21** [comment]
@lamping7 yes I am aware of the ROS2 Ardent case, but it's still being used in production in our FastRTPS bridge I want to keep it for at least 1 more month. I will deprecate it myself.

**Comment by TSC21** [comment]
The raspi has a different toolchain from armhf for cross compilation.

**Comment by TSC21** [comment]
And finally: yeah if you don't mind, can you add the ROS2 images to the Makefile? Thanks

File: docker/px4-dev/README.md [hunk 2]
```diff
@@ -6,14 +6,22 @@ License: according to LICENSE.md in the root directory of the PX4 Firmware repos
...
+ - px4-dev-base-archlinux: Arch Linux latest
+ - px4-dev-base-bionic: Ubuntu 18.04
+   - px4-dev-armhf
    - px4-dev-clang
      - px4-dev-nuttx-clang
+   - px4-dev-ecl
+   - px4-dev-nuttx
+   - px4-dev-raspi
+   - px4-dev-simulation: Gazebo 9
+     - px4-dev-ros: ROS Melodic
```
Comment (lamping7) [inline docker/px4-dev/README.md]: -> `ros-melodic`

**Comment by lamping7** [comment]
Yup. I might as well get base-xenial in Jenkins too

**Comment by lamping7** [comment]
This is good now. I removed all the `ls` lines from Jenkins while I was in there and decreased builds kept. Of course Jenkins can't pull the container until I fix docker hub, but I don't want to pull the trigger until we're all happy.

**Comment by dagar** [comment]
> Additional discussion points:
> 
> * Keep raspi? We have armhf.

Needs someone to test (and care).

> * Why is everything named "-dev"? It doesn't really distinguish it from anything else.

There was going to be a distinction between development containers and containers for running SITL that never quite materialized.

**Pull request merged**